odoo/service/server.py

```diff
@@ -435,6 +434,57 @@ def target():
             t.start()
             _logger.debug("cron%d started!" % i)
 
+    def async_thread(self, n):
+        from odoo.addons.base.models.ir_async import IrAsync
+        conn = odoo.sql_db.db_connect('postgres')
+
+        with conn.cursor() as cr:
+            pg_conn = cr._cnx
+            cr.execute("LISTEN odoo_async")
+            cr.commit()
+
+            # We assume all databases are notified at startup so we poll
+            # them all once for leftover jobs. When all leftover jobs
+            # are processed, we wait for postgres notifications
+            notified_dbs = [
+                dbname
+                for dbname, registry
+                in odoo.modules.registry.Registry.registries.d.items()
+                if registry.ready
+            ]
+
+            while True:
+
+                # avoid thunderhead effect: On NOTIFY, all workers are
+                # awaken at the same time, the sleep prevents they all
+                # poll the database at the exact same time.
+                time.sleep(n / 100 % .1)
+
+                for dbname in notified_dbs:
+                    _logger.debug("async%d processing db %s", n, dbname)
+                    threading.currentThread().start_time = time.time()
+                    IrAsync._process_jobs(dbname)
+                threading.currentThread().start_time = None
+                _logger.debug("async%d awaiting signal", n)
+                select.select([pg_conn], [], [], SLEEP_INTERVAL)
+                pg_conn.poll()
+
+                # The NOTIFY payload holds the database name where a job
+                # got enqueued
+                notified_dbs = {notify.payload for notify in pg_conn.notifies}
```

Comment:
Where is `notified_dbs` used ?
Edit: it should probably be self.notified_dbs like below :thinking:

Reply:
mind the loop, it is actually used above